addons/web/controllers/main.py

```diff
@@ -506,8 +506,20 @@ def binary_content(xmlid=None, model='ir.attachment', id=None, field='datas', un
 
     # get content after cache control
     if model == 'ir.attachment' and obj.type == 'url' and obj.url:
-        status = 301
+        old_status, status = status, 301
         content = obj.url
+
+        url_match = re.match("^/(\w+)/(.+)$", obj.url)
+        if url_match:
+            module = url_match.group(1)
+            module_path = get_module_path(module)
+            module_ressource_path = get_resource_path(module, url_match.group(2))
+            if module_path and module_ressource_path:
+                module_path = os.path.normpath(module_path)
+                module_ressource_path = os.path.normpath(module_ressource_path)
+                if os.path.commonprefix([module_path, module_ressource_path]) == module_path:
+                    status = old_status
```

Comment:
Which part? status = old_status is required otherwise we will have a 301 with the file data as content.